moveit_core/kinematics_base/include/moveit/kinematics_base/kinematics_base.h

```diff
@@ -154,6 +154,9 @@ class MOVEIT_KINEMATICS_BASE_EXPORT KinematicsBase
   using IKCallbackFn = std::function<void(const geometry_msgs::msg::Pose&, const std::vector<double>&,
                                           moveit_msgs::msg::MoveItErrorCodes&)>;
 
+  /** @brief Signature for a cost function used to evaluate IK solutions. */
+  using IKCostFn = std::function<double(const geometry_msgs::msg::Pose&, const moveit::core::RobotState&)>;
```

Comment:
Will this be enough to cover our use cases? I thought we might want to have the seed state in here as well

Reply:
I think the seed state is a good idea to put in here as well. For our use case, there's actually [a lot more](https://github.com/PickNikRobotics/bio_ik/blob/ad75b2e8085338311a2f6e7224f43b9b1f35a79c/include/bio_ik/goal.h#L49-L95) that can go into evaluating a cost function. While including all of this here is probably superfluous, seed state will allow us to do things like a minimal displacement cost function

Reply:
I also added a `JointModelGroup*` parameter, since a lot of what we can do with `RobotState` requires it

Reply:
One thing to note is that because this is a `std::function` if you have the state externally you want to pass in there is always the option of using a lambda and capturing those values.